src/JsonRepositoryWriter.php

```diff
@@ -91,27 +91,30 @@ public function save(): void
             }
             $data['includes'][] = $content;
         }
+        unset($tool, $content);
 
         foreach ($this->plugins as $plugin) {
             if (null === $content = $this->processPlugin($plugin)) {
                 continue;
             }
             $data['includes'][] = $content;
         }
+        unset($plugin, $content);
 
         $this->dumpFile('repository.json', $data);
+        unset($data);
 
         $flipped = array_flip($this->dumpedFileNames);
         foreach (glob($this->baseDir . '/*') as $filename) {
-            if (!array_key_exists($filename, $flipped)) {
+            if (!is_dir($filename) && !array_key_exists($filename, $flipped)) {
                 $this->filesystem->remove($filename);
             }
         }
```

Comment:
This would leave empty directories, wouldn't it? They should be removed also if no files are in it anymore.

Reply:
Yes, this is correct - I left them in for the moment as they will not be committed anyway (git ignores them).
We can remove them however.